packages/aztec.js/src/note/index.js

```diff
@@ -1,171 +1,168 @@
+     *
+     * @param {string} publicKey hex-formatted public key
+     * @param {string} viewingKey hex-formatted viewing key
+     * @param {string} owner Ethereum address of note owner
+     * @param {Object} setupPoint trusted setup point
      */
-    this.owner = owner;
-    if (publicKey) {
-        if (typeof publicKey !== 'string') {
-            throw new Error(`expected key type ${typeof publicKey} to be of type string`);
-        }
-        if (publicKey.length !== 200) {
-            throw new Error(`invalid public key length, expected 200, got ${publicKey.length}`);
+    constructor(publicKey, viewingKey, owner = '0x', setupPoint) {
+        if (publicKey && viewingKey) {
+            throw new Error('expected one of publicKey or viewingKey, not both');
         }
         /**
-         * Viewing key of note. BN instance in bn128 group's reduction context
-         * @member {BN}
-         */
-        this.a = null;
-        /**
-         * Value of note. BN instance in bn128 group's reduction context
-         * @member {BN}
+         * Ethereum address of note's owner
+         * @member {string}
          */
-        this.k = null;
-        /**
-         * AZTEC commitment point \gamma, a bn128 group element
-         * @member {Point}
-         */
-        this.gamma = bn128.curve.decodePoint(publicKey.slice(2, 68), 'hex');
-        /**
-         * AZTEC commitment point \sigma, a bn128 group element
-         * @member {Point}
-         */
-        this.sigma = bn128.curve.decodePoint(publicKey.slice(68, 134), 'hex');
-        /**
-         * Note's ephemeral key, a secp256k1 group element. A note owner can use this point
-         * to compute the note's viewing key.
-         * @member {Point}
-         */
-        this.ephemeral = secp256k1.ec.keyFromPublic(publicKey.slice(134, 200), 'hex');
-    }
-    if (viewingKey) {
-        if (typeof viewingKey !== 'string') {
-            throw new Error(`expected key type ${typeof viewingKey} to be of type string`);
+        this.owner = owner;
+        if (publicKey) {
+            if (typeof publicKey !== 'string') {
+                throw new Error(`expected key type ${typeof publicKey} to be of type string`);
+            }
+            if (publicKey.length !== 200) {
+                throw new Error(`invalid public key length, expected 200, got ${publicKey.length}`);
+            }
+            /**
+             * Viewing key of note. BN instance in bn128 group's reduction context
+             * @member {BN}
+             */
+            this.a = null;
+            /**
+             * Value of note. BN instance in bn128 group's reduction context
+             * @member {BN}
+             */
+            this.k = null;
+            /**
+             * AZTEC commitment point \gamma, a bn128 group element
+             * @member {Point}
+             */
+            this.gamma = bn128.curve.decodePoint(publicKey.slice(2, 68), 'hex');
```

Comment:
It might be helpful to describe the structure of `publicKey` in the comments (e.g. 1st 2 characeters are `0x`, next 128 are two elliptic curve points that encrypt the note value, with the `ephemeral` key taking up the rest of the space, which is used by the note owner to decrypt their note.